### PR TITLE
Add tool to connect PS and common repos

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

This PR adds

1. A "tools/DevTools" folder to put tools for developing/debugging purpose
2. A script module to connect PS repo and common repo, following the steps in [engineering hub](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azps_docs/implementation#how-to-debug-azure-azure-powershell-common)
3. A README.md which describes the script

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by isra-fel** [comment]
Todo: update doc on engineering hub after this is merged

**Review by BethanyZhou — COMMENTED** [review]
This is a really cool script! Does this script support connecting other module with common repo? How about I want to debug Az.KeyVault and common repo

File: tools/DevTools/CommonRepo.psm1 [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+
+  Write-Host "2/2: Remove the dependency of those common projects from .targets file"
+
+  $Patterns = @(
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Authentication.Abstractions"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Common"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Clients.Graph.Rbac"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Clients.ResourceManager"',
+    '<PackageReference Include="Microsoft.Azure.PowerShell.Common.Share"'
```
Comment (isra-fel) [inline tools/DevTools/CommonRepo.psm1:64]: These 5 projects might not be enough in some cases, for example when debugging with Az.Compute.
I should consider adding all common projects in the script.

File: tools/DevTools/CommonRepo.psm1 [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  Push-Location "$RepoRoot/src/Accounts"
+  try {
+    foreach ($csproj in $csprojs) {
+      $csproj = [System.IO.Path]::Combine($CommonRepoPath, $csproj)
+      $csproj = [System.IO.Path]::GetFullPath($csproj)
+      dotnet sln add $csproj
+      if ($LASTEXITCODE -ne 0) {
+        throw "Failed to add $csproj to Accounts.sln"
+      }
+      dotnet add ./Accounts/Accounts.csproj reference $csproj
```
Comment (isra-fel) [inline tools/DevTools/CommonRepo.psm1:41]: How do I update the script to support connecting common repo to not only Az.Accounts but also service modules like Az.KeyVault?

File: tools/DevTools/CommonRepo.psm1 (hunk 1 above)
Comment (isra-fel) [reply tools/DevTools/CommonRepo.psm1]: Now all the common projects are added.

File: tools/DevTools/CommonRepo.psm1 (hunk 2 above)
Comment (isra-fel) [reply tools/DevTools/CommonRepo.psm1]: Consider `Connect-CommonRepo -Solution ./src/KeyVault/KeyVault.sln -Csproj ./src/KeyVault/KeyVault.csproj`. By default Solution=Accounts.sln, -Csproj=Authentication.proj

**Comment by isra-fel** [comment]
Hi @msJinLei since you are updating common repo, will you be able to try my script and give any feedback if possible? Thanks in advance.

**Comment by msJinLei** [comment]
> Hi @msJinLei since you are updating common repo, will you be able to try my script and give any feedback if possible? Thanks in advance.

Sorry for late response. Just see the email. The script works well for me.

**Comment by BethanyZhou** [comment]
Hi @isra-fel , I hit an issue in  when using this script to connect Az.Accounts and common repo.

Here is the error msg:
```
[assembly: AssemblyCopyright(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyCopyright)]
```


```
The type or namespace name 'WindowsAzure' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?)	AuthenticationAssemblyLoadContext	C:\Azure\fork\azure-powershell\src\Accounts\AuthenticationAssemblyLoadContext\Properties\AssemblyInfo.cs	25	Active
```

**Comment by isra-fel** [comment]
Thank you @BethanyZhou . How did you fix the issue?

**Review by YanaXu — APPROVED** [review]

File: tools/DevTools/CommonRepo.psm1 (hunk 2 above)
Comment (BethanyZhou) [reply tools/DevTools/CommonRepo.psm1]: Does current script support connecting service modules?

**Pull request merged**